src/bin/websurfx.rs

```diff
@@ -6,6 +6,11 @@
 use std::net::TcpListener;
 use websurfx::{config::parser::Config, run};
 
+/// A dhat heap memory profiler
+#[cfg(feature = "dhat-heap")]
+#[global_allocator]
+static ALLOC: dhat::Alloc = dhat::Alloc;
```

Comment:
It seems that we haven't added `dhat` to the dependencies, but that's okay for this PR. I noticed that we did that in another part of this PR series.

Reply:
Yes, my bad :sweat_smile:  but I think when I split the PRs I just missed it, but I think let's have that be it in the next PR. :slightly_smiling_face: What do you say?